### DISCUSSION TITLE
Get the branch or tag ref on delete

### DISCUSSION THREAD

**Discussion by izolate** [community/community#26496] [Actions]
<p>Is it possible to get the branch name or tag ref on delete?</p>
<pre><code>on:
  delete

jobs:
  foo:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v1
      - run: echo ${GITHUB_REF}
</code></pre>
<p>As per the docs, $GITHUB_REF resolves to the “default branch”, which in my case is master. I’m struggling to see how returning the default branch is more useful than resolving the actual branch or tag ref that was deleted.</p>
<p>Is there any way to get the deleted branch name?</p>

**Comment by chrispat** [comment, marked as answer]
<p>The event payload for delete has this infoarmtion <a href="https://developer.github.com/v3/activity/events/types/#deleteevent" rel="nofollow noopener">https://developer.github.com/v3/activity/events/types/#deleteevent</a>.</p>
<p>You can use the github context to access data in that payload <a href="https://help.github.com/en/github/automating-your-workflow-with-github-actions/contexts-and-expression-syntax-for-github-actions#github-context" rel="nofollow noopener">https://help.github.com/en/github/automating-your-workflow-with-github-actions/contexts-and-expression-syntax-for-github-actions#github-context</a></p>

**Comment by mukunku** [comment]
<p>Just to add to this for people like me who didn’t know how to access the ref and ref_type properties for the event this is how you do it in the yaml:</p>
<pre><code>- name: Hello
  run: echo "${{ github.event.ref }} - ${{ github.event.ref_type }}"
</code></pre>
